src/status_im/communities/core.cljs

```diff
@@ -131,18 +131,29 @@
                db
                chat-ids)})
 
-(rf/defn handle-community
-  [{:keys [db]} {:keys [id] :as community}]
-  (when id
-    {:db (assoc-in db [:communities id] (<-rpc community))}))
+(rf/reg-event-fx
+ :communities/handle-community
+ (fn [{:keys [db]}
+      [{token-permissions :tokenPermissions
+        :keys             [token-permissions-check joined id]
+        :as               community}]]
+   (when id
+     (let [has-token-permissions? (not (seq token-permissions))]
+       (cond-> {:db (assoc-in db [:communities id] (<-rpc community))}
+         (and has-token-permissions? (not joined))
+         (update :fx conj [:dispatch [:chat.ui/spectate-community id]])
+         (and has-token-permissions? (nil? token-permissions-check))
+         (update :fx conj [:dispatch [:communities/check-permissions-to-join-community id]])
+         (and has-token-permissions? (not (get-in db [:community-channels-permissions id])))
+         (update :fx conj [:dispatch [:communities/check-all-community-channels-permissions id]]))))))
 
 (rf/defn handle-communities
   {:events [::fetched]}
   [{:keys [db]} communities]
-  {:db (reduce (fn [db {:keys [id] :as community}]
-                 (assoc-in db [:communities id] (<-rpc community)))
-               db
-               communities)})
+  {:fx
+   (->> communities
+        (map :id)
+        (map #(vector :dispatch [:communities/handle-community %])))})
```

Comment:
Is this ok? @flexsurfer @ilmotta 
It's used in process-response
I saw `:fx` in `mergeable-keys` but I'm not sure if this works with `rf/merge`

Reply:
why that's a lot of dispatches, why would you want to do that change?

Reply:
Because these events are previously triggered by UI at component start
and I remember re-frame suggests don't trigger in UI if you already got events to trigger the UI

this also improves user exp and fewer renders, since data already loaded when user enters the community overview page